File: 20191009/swea_1231_중위순회.py
import sys;sys.stdin=open('1231.txt')

def inorder(v):
    global answer
    if v == 0: return
    inorder(L[v])
    answer += words[v]
    inorder(R[v])
    
for t_case in range(1, 11):
    V = int(input())
    E = V - 1
    L = [0] * (V+1)
    R = [0] * (V+1)
    words = ['0']
    for i in range(1, V+1):
        arr = list(input().split())
        if len(arr) == 4:
            p, w, l, r = int(arr[0]), arr[1], int(arr[2]), int(arr[3])
        elif len(arr) == 3:
            p, w, l, r = int(arr[0]), arr[1], int(arr[2]), 0
        else:
            p, w, l, r = int(arr[0]), arr[1], 0, 0
        L[p], R[p] = l, r
        words.append(w)
    answer = ''
    inorder(1)
    print("#{} {}".format(t_case, answer))

File: 2019_late/20191009/swea_1231_중위순회.c
#include <stdio.h>

int L[200];
int R[200];
char words[200];

void inorder(int v)
{
    if (v==0){
        return;
    }
    inorder(L[v]);
    printf("%c", words[v]);
    inorder(R[v]);
}

int main()
{
    for (int t_case = 1;t_case < 11; t_case++)
    {
        int V,p,l,r;      
        char w;
        scanf("%d",&V);
        for (int i = 1; i <= V; i++)
        {
            scanf("%d %c",&p, &w);
            words[p] = w;
            if ((p * 2) + 1 <= V)
            {
                scanf("%d %d", &L[p], &R[p]);
            }
            else if (p * 2 <= V) 
            {
                scanf("%d", &L[p]);
            }
        }
        printf("#%d ", t_case);
        inorder(1);
        printf("\n");
        for (int i = 0; i < V + 1; i++){
            words[i] = '0';
            L[i] = 0;
            R[i] = 0;
        }
        
    }
    return 0;
}